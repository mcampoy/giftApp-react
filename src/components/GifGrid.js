import React from 'react';
import { useFetchGifts } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

const GifGrid = ( {category} ) => {

    const {data:gifs, loading} = useFetchGifts( category );

    return (
        <>
            <h6 className="animate__animated animate__fadeIn"> Estos son los resultados sobre "{category}" </h6>

            { loading && <p className=" loading animate__animated animate__flash" >cargando...</p> }

            <div  className="card-grid animate__animated">

                {
                gifs.map(  gif => (
                    <GifGridItem 
                    key={gif.id}
                    { ...gif }
                    />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;