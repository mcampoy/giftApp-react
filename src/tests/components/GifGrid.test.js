import React from 'react'
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifts } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid />', () => {

    const category = 'cats';
    
    test('debe hacer match con el componente', () => {

        useFetchGifts.mockReturnValue({
            data:[],
            loading: true
        })

        const wrapper = shallow( <GifGrid category={category} /> )
        expect(wrapper).toMatchSnapshot(); 
    });
    
    test('debe mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/test/1.jpg',
            title: 'Test'
        },
        {
            id: '123',
            url: 'https://localhost/test/1.jpg',
            title: 'Test'
        }];


        useFetchGifts.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={category} /> );
        
        // expect(wrapper).toMatchSnapshot(); 
        expect (wrapper.find('p').exists()).toBe(false);
        expect( wrapper.find('GifGridItem').length).toBe(gifs.length)

    })
    


})
