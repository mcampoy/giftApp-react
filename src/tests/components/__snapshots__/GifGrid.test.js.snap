// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> debe hacer match con el componente 1`] = `
<Fragment>
  <h6
    className="animate__animated animate__fadeIn"
  >
     Estos son los resultados sobre "
    cats
    " 
  </h6>
  <p
    className=" loading animate__animated animate__flash"
  >
    cargando...
  </p>
  <div
    className="card-grid animate__animated"
  />
</Fragment>
`;
